lib/algoliax/resources/index.ex

```diff
@@ -87,17 +109,24 @@ defmodule Algoliax.Resources.Index do
   end
 
   def delete_index(module, settings) do
-    request(%{action: :delete_index, url_params: [index_name: index_name(module, settings)]})
+    index_name(module, settings)
+    |> Enum.map(fn index_name ->
+      request(%{action: :delete_index, url_params: [index_name: index_name]})
+    end)
+    |> case do
+      [single_result] -> single_result
+      [_ | _] = multiple_result -> multiple_result
```

Comment:
We have a lot of duplicated code for this, I guess you want to unwrap the result in case it's a one-index usecase?

I would recommend having this in the `utils` module.